src/routes/safe/store/actions/transactions/fetchTransactions/loadAllTransactions.ts

```diff
@@ -0,0 +1,81 @@
+import { List, Map } from 'immutable'
+import axios, { AxiosResponse } from 'axios'
+
+import { getAllTxServiceUriTo, getTxServiceHost } from 'src/config'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import { Transaction } from '../../../models/types/transactions'
+
+type ServiceUriParams = {
+  safeAddress: string
+  limit?: number
+  offset?: number
+  orderBy?: string // todo: maybe this should be key of MultiSigTransaction | keyof EthereumTransaction
+  queued?: boolean
+  trusted?: boolean
+}
+
+type EndpointResponse = {
+  count: number
+  next?: string
+  previous?: string
+  results: Transaction[]
+}
+
+const getAllTransactionsUrl = (safeAddress: string) => {
+  const host = getTxServiceHost()
+  const address = checksumAddress(safeAddress)
+  const base = getAllTxServiceUriTo(address)
+
+  return `${host}${base}`
+}
+
+const fetchAllTransactions = async (
+  urlParams: ServiceUriParams,
+  eTag: string | null,
+): Promise<{ eTag: string; results: Transaction[] }> => {
+  const { safeAddress, limit, offset, orderBy, queued, trusted } = urlParams
+  try {
+    const url = getAllTransactionsUrl(safeAddress)
+
+    const config = {
+      params: {
+        limit,
+        offset,
+        orderBy,
+        queued,
+        trusted,
+      },
+      headers: eTag ? { 'If-None-Match': eTag } : undefined,
+    }
+
+    const response: AxiosResponse<EndpointResponse> = await axios.get(url, config)
+
+    if (response.data.count > 0) {
```

Comment:
can we change it to `if(!response.data.count) return`? That way we avoid nested conditions

Reply:
I used that approach because we would need to return the default values: `return { eTag, results: [] }` and didn't want to repeat the code